diff --git a/build/make/tools/releasetools/add_img_to_target_files.py b/build/make/tools/releasetools/add_img_to_target_files.py
index 8479eb03d..14e76175e 100755
--- a/build/make/tools/releasetools/add_img_to_target_files.py
+++ b/build/make/tools/releasetools/add_img_to_target_files.py
@@ -61,6 +61,9 @@ import common
 import rangelib
 import sparse_img
 
+# SHRP must be downloaded + AVB hash added BEFORE using it here
+recovery_file = "/tmp/do_NOT_flash_shrp_latest_NO-AVB-HASH.img"
+
 if sys.hexversion < 0x02070000:
   print("Python 2.7 or newer is required.", file=sys.stderr)
   sys.exit(1)
@@ -817,6 +820,10 @@ def AddImagesToTargetFiles(filename):
     partitions['recovery'] = os.path.join(
         OPTIONS.input_tmp, "IMAGES", "recovery.img")
 
+    # overwrite with latest SHRP image
+    banner("overwrite recovery with custom one")
+    partitions['recovery'] = recovery_file
+
     if not os.path.exists(partitions['recovery']):
       recovery_image.WriteToDir(OPTIONS.input_tmp)
       if output_zip:
